python/pyspark/sql/functions.py

```diff
@@ -2776,6 +2787,40 @@ def regexp_extract(str, pattern, idx):
     return Column(jc)
 
 
+def regexp_extract_all(str, pattern, idx):
+    r"""Extract all matches of the given group in a regex, from the specified string column.
+    If the regex did not match, or the specified group did not match, an empty array is returned.
+
+    .. versionadded:: 3.1.0
```

Comment:
Let's target 3.2.0

Reply:
Oh, I thought this was the version where the Scala function was introduced. Changed.